packages/nimble-components/src/combobox/index.ts

```diff
@@ -159,11 +159,11 @@ export class Combobox
             const prevSelectedValue = this.options[this.selectedIndex]?.text;
             const nextSelectedValue = this.options[selectedIndex]?.text;
 
-            this.selectedIndex = prevSelectedValue !== nextSelectedValue
-                ? selectedIndex
-                : this.selectedIndex;
+            if (prevSelectedValue !== nextSelectedValue) {
+                this.selectedIndex = selectedIndex;
+            }
```

Comment:
No need to reassign the same value. This is a little clearer in terms of intent.